drivers/sensor/st/stm32_vbat/stm32_vbat.c

```diff
@@ -116,9 +121,13 @@ static int stm32_vbat_init(const struct device *dev)
 	return 0;
 }
 
+#define STM32_VBAT_GET_ADC_OR_NULL(inst)                                                           \
+	COND_CODE_1(DT_NODE_HAS_STATUS(DT_INST_IO_CHANNELS_CTLR(inst), okay),                      \
+		    (DEVICE_DT_GET(DT_INST_IO_CHANNELS_CTLR(inst))), (NULL))
```

Comment:
What about
```suggestion
	IF_ENABLED(DT_NODE_HAS_STATUS_OKAY(DT_INST_IO_CHANNELS_CTLR(inst)),                      \
		               (DEVICE_DT_GET(DT_INST_IO_CHANNELS_CTLR(inst))))
```

Reply:
We do not have `DT_NODE_HAS_STATUS_OKAY` API

Reply:
I've seen the other PR indeed. Strange.. Thanks for having had a try anyway

Reply:
https://github.com/zephyrproject-rtos/zephyr/blame/main/soc/nxp/rw/soc.c#L293 @decsny it looks like it won't work.